### PR TITLE
feat(core): Add missing core features

### LINKED ISSUE #3: Missing core features
Right now drift is not complete at all, there are still a bunch of features which it cannot migrate.

Here is a list of it, so we know what we need to work on before having "complete" refactor. (if something is missing, just edit the list and add it or post a comment and I'll add it)

- [ ] Data providers
- [ ] Exception throwing
- [ ] Helper methods
- [ ] Skipping tests
- [ ] setUp/tearDown
- [ ] beforeClass/afterClass
- [ ] Migrate traits to uses statements

### PR DESCRIPTION
**Pull request by olivernybroe**
This PR is for adding the missing core features.

- [x] Data providers
- [x] Exception throwing
- [x] Helper methods
- [x] Skipping tests
- [x] setUp/tearDown
- [x] beforeClass/afterClass
- [x] Migrate traits to uses statements

Bugfixes:
- [x] Fix using helper methods in non tests, should not use `$this`
- [x] Fix adding `uses` call, if class is extending a custom test case
- [x] Fix overriding methods, not getting converted to helper methods

closes #3

### CODE REVIEW

**Comment by nunomaduro** [comment]
@olivernybroe This is looking so good. Can you update this to use `pest 0.2`?

**Comment by olivernybroe** [comment]
@nunomaduro Sure, I'll do it as part of this pr then 👍

Really like how easy it is to have test for this.

File: composer.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
     "type": "library",
     "require": {
         "php": "^7.3",
-        "rector/rector": "^0.7.26"
+        "rector/rector": "^0.7.26",
+        "pestphp/pest": "^0.2@dev"
```
Comment (nunomaduro) [inline composer.json:8]:
```suggestion
        "pestphp/pest": "^0.2"
```

Comment (olivernybroe) [reply composer.json]: It wouldn't let me do it before :) but seemed to work this time 👍

File: tests/fixtures/PHPUnit/phpunit_trait_to_pest_uses.php.inc [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    }
+}
+?>
+-----
+<?php
+use Pest\Drift\Testing\fixtures\TraitA;
+use PHPUnit\Framework\TestCase;
+uses(TraitA::class);
+it('testUses', function () {
+    $this->assertTrue(true);
```
Comment (nunomaduro) [inline tests/fixtures/PHPUnit/phpunit_trait_to_pest_uses.php.inc:21]: Can we use just `assertTrue(true);` instead?

Comment (olivernybroe) [reply tests/fixtures/PHPUnit/phpunit_trait_to_pest_uses.php.inc:21]: Yes, but that will be the job of another rector #4. 

So this rector is just for converting to pest in the most simple way, there will then be other rectors for cleaning up to the most pest like syntax 👍 

Ofc. the tool we are creating here will run all of the rectors automatically.

**Comment by nunomaduro** [comment]
👏 👏 👏 👏 👏 👏

**Pull request merged**